cloudinit/utils.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudinit
+
+import (
+	"encoding/base64"
+	"strings"
+	"text/template"
+)
+
+const (
+	rootOwnerValue = "root:root"
+)
+
+var (
+	defaultTemplateFuncMap = template.FuncMap{
+		"Base64Encode": templateBase64Encode,
+		"Indent":       templateYAMLIndent,
+	}
+)
+
+func templateBase64Encode(s string) string {
+	return base64.StdEncoding.EncodeToString([]byte(s))
+}
+
+func templateYAMLIndent(i int, input string) string {
```

Comment:
this doesn't look used in any of the templates, could nix it and the tests entirely.

Reply:
Should be used in here https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/pull/21/files/8cfb40632069cce723288772ae1d1c71d6d07ac3#diff-9b1997df3b911165aa3156f18e00d159R24

Reply:
ah yes, so it is. my, fault